### PR TITLE
fix(Osv): Make the configuration optional again

### PR DESCRIPTION
**Pull request by fviernau**
See individual commits.

### CODE REVIEW

File: model/src/main/kotlin/config/AdvisorConfiguration.kt [hunk 1]
```diff
@@ -50,7 +50,7 @@ data class GitHubDefectsConfiguration(
      * The access token to authenticate against the GitHub GraphQL endpoint.
      */
     @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
-    val token: String?,
+    val token: String? = null,
```
Comment (sschuberth) [inline model/src/main/kotlin/config/AdvisorConfiguration.kt:53]: I believe this property is not optional, and should not have been nullable to start with. Can you confirm @oheger-bosch?

File: model/src/main/kotlin/config/AdvisorConfiguration.kt [hunk 2]
```diff
@@ -155,11 +155,11 @@ data class VulnerableCodeConfiguration(
...
      */
-    val serverUrl: String?,
+    val serverUrl: String? = null,
 
     /**
      * The optional API key to use.
      */
     @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
-    val apiKey: String?
+    val apiKey: String? = null
```
Comment (sschuberth) [inline model/src/main/kotlin/config/AdvisorConfiguration.kt:164]: I believe the comment is outdated and this property actually is not optional anymore. @oheger-bosch, do you know if also self-hosted VulnerableCode instances require an API key by now?

File: advisor/src/main/kotlin/advisors/Osv.kt [hunk 3]
```diff
@@ -58,7 +58,7 @@ class Osv(name: String, config: OsvConfiguration) : AdviceProvider(name) {
     class Factory : AbstractAdviceProviderFactory<Osv>("OSV") {
         override fun create(config: AdvisorConfiguration) =
             // OSV does not require any dedicated configuration to be present.
-            Osv(type, config.forProvider { osv })
+            Osv(type, config.forProvider { osv ?: OsvConfiguration() })
```
Comment (sschuberth) [inline advisor/src/main/kotlin/advisors/Osv.kt:61]: Sorry for breaking this. Looks like this was broken also for `OssIndex` (from where I took the code) before. Mind finding it for `OssIndex`, too?

**Review by sschuberth — CHANGES_REQUESTED** [review]

**Comment by sschuberth** [comment]
As a side-note, I believe the `AdvisorConfiguration` to be quite messy, as they have a wild mix of `null` default values (and some other place applies the actual default value) vs. having the real default value in the configuration class itself. I believe the latter should be the approach to align on, as otherwise the used config as written to the ORT result file as part of an `AdvisorRun` doe snot contain the actually used values. This is probably something to keep in mind for https://github.com/oss-review-toolkit/ort/pull/6613.

File: model/src/main/kotlin/config/AdvisorConfiguration.kt (hunk 1 above)
Comment (fviernau) [reply model/src/main/kotlin/config/AdvisorConfiguration.kt:53]: Can this be considered out of scope of this commit / PR?

File: advisor/src/main/kotlin/advisors/Osv.kt (hunk 3 above)
Comment (fviernau) [reply advisor/src/main/kotlin/advisors/Osv.kt:61]: Done.

**Review by sschuberth — APPROVED** [review]

**Pull request merged**